### PR TITLE
add EMail endpoints from gitlab api

### PR DESCRIPTION
**Pull request by ulrichSchreiner**
hi,

i've implemented some REST-endpoints for the management of user emails. The functions should be identical to the `SSH-Key` functions

### CODE REVIEW

File: users.go [hunk 1]
```diff
@@ -437,3 +437,152 @@ func (s *UsersService) UnblockUser(user int) error {
 		return fmt.Errorf("Received unexpected result code: %d", resp.StatusCode)
 	}
 }
+
+// EMail represents an EMail.
```
Comment (svanharmelen) [inline users.go]: Could you rename all instances of `EMail` to `Email` please?

File: users.go [hunk 2]
```diff
@@ -437,3 +437,152 @@ func (s *UsersService) UnblockUser(user int) error {
...
+//
+// GitLab API docs: https://doc.gitlab.com//help/api/users.md#list-emails
```
Comment (svanharmelen) [inline users.go]: This URL is not correct, should be: `http://doc.gitlab.com/ce/api/users.html#list-emails`

File: users.go [hunk 3]
```diff
@@ -437,3 +437,152 @@ func (s *UsersService) UnblockUser(user int) error {
...
+// ListEMails gets a list of currently authenticated user's EMails.
+//
+// GitLab API docs: http://doc.gitlab.com/ce/api/users.html#list-emails
+func (s *UsersService) ListEMails() ([]*EMail, *Response, error) {
+	req, err := s.client.NewRequest("GET", "user/emails", nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var k []*EMail
```
Comment (svanharmelen) [inline users.go]: The `k` stood for `key` in the `SSH` calls, so it should be `e` here to keep things consistent (applies to all instances of course)

File: users.go [hunk 4]
```diff
@@ -437,3 +437,152 @@ func (s *UsersService) UnblockUser(user int) error {
...
+
+	return k, resp, err
+}
+
+// ListEMailsForUser gets a list of a specified user's EMails. Available
+// only for admin
+//
+// GitLab API docs:
+// http://doc.gitlab.com/ce/api/users.html#list-emails-for-user
+func (s *UsersService) ListEMailsForUser(user int) ([]*EMail, *Response, error) {
```
Comment (svanharmelen) [inline users.go]: I would prefer `uid` over `user` (applies to all instances of course)

File: users.go [hunk 5]
```diff
@@ -437,3 +437,152 @@ func (s *UsersService) UnblockUser(user int) error {
...
+	return e, resp, err
+}
+
+// DeleteEMail deletes email owned by currently authenticated user. This is an
+// idempotent function and calling it on a key that is already deleted or not
+// available results in 200 OK.
+//
+// GitLab API docs:
+// http://doc.gitlab.com/ce/api/users.html#delete-email-for-current-owner
+func (s *UsersService) DeleteEMail(kid int) (*Response, error) {
```
Comment (svanharmelen) [inline users.go]: `kid` should be `eid` to keep things consistent (applies to all instances of course)

**Comment by ulrichSchreiner** [comment]
hi, 

thanx for the comments. i implemented them. hope it's ok now.

usc

**Comment by svanharmelen** [comment]
@ulrichSchreiner looks great now, thanks!

**Pull request merged**